store/src/db/mod.rs

```diff
@@ -130,4 +131,36 @@ impl Db {
             .await
             .map_err(|_| anyhow!("Get account hashes task failed with a panic"))?
     }
+
+    pub async fn apply_block(
+        &self,
+        allow_acquire: oneshot::Sender<()>,
+        acquire_done: oneshot::Receiver<()>,
```

Comment:
Let's add doc comments to this function explaining the purpose of `allow_acquire` and `acquire_done` (also, maybe there are more self-explanatory names we can use here?).

Also, I'm guessing it is not possible to return `transaction` from this function - right? (if it is possible, we could return the transaction to the caller, and then the caller would call `transaction.commit()` when needed).

Reply:
> also, maybe there are more self-explanatory names we can use here?

Suggestions?

> Also, I'm guessing it is not possible to return transaction from this function - right?

transaction is `!Send` and `!Sync` [ref](https://docs.rs/rusqlite/latest/rusqlite/struct.Transaction.html#synthetic-implementations). So it can not be moved across threads.

Added docs

Reply:
For names, maybe we can use something like `db_update_completed` instead of `allow_acquire` and `mem_update_completed` instead of `acquire_done`?

Reply:
I think these are not better names.

`db_update_completed` at least to me implies the transactions was commited which is not the case here (it happens after both events).
`mem_update_complete` to me implies the in-memory representation matches the DB with the new data. In here the update is about to start.